initrd/bin/kexec-save-default

```diff
@@ -129,6 +129,41 @@ if [ ! -r $ENTRY_FILE -o ! -r $HASH_FILE ]; then
 	die "Failed to write default config"
 fi
 
+# logic to parse OS initrd to get extract crypttab, its filepaths and its OS defined options
+mkdir -p /tmp/initrd_extract
+cd /tmp/initrd_extract
+# User just selected a default boot option. Get initrd filename
+current_default_initrd=$(cat /boot/kexec_default_hashes.txt | grep initr | awk -F " " {'print $NF'} | sed 's/\.\//\/boot\//g')
+# Get crypttab files paths from initrd
+crypttab_files=$(zcat $current_default_initrd | cpio --list --quiet | grep crypttab) > /dev/null || crypttab_files=$(cpio --list --quiet < $current_default_initrd | grep crypttab) > /dev/null
+
+if [ ! -z "$crypttab_files" ]; then
+	echo "Extracting current selected default boot's $current_default_initrd for crypttab analysis..."
+	zcat $current_default_initrd | cpio -id --quiet "$crypttab_files" > /dev/null || cpio -id --quiet < $current_default_initrd "$crypttab_files" > /dev/null
+
+	crypttab_entries=$(cat "$crypttab_files"|uniq| sed 's/none/\/secret.key/g')
+	cd - > /dev/null
+
+	#insert current default boot's initrd crypttab location into tracking file to be overwritten into initramfs
+	echo "The following OS crypttab was found in intrd and Heads will overwrite it: $crypttab_files"
+	echo "Adding found crypptab files location under $bootdir/kexec_initrd_crypttab_path.txt"	
+	echo "$crypttab_files" > "$bootdir/kexec_initrd_crypttab_path.txt"
+else
+	echo "No crypttab file found in extracted initrd. Removing $bootdir/kexec_initrd_crypttab_path.txt"
+	rm -f "$bootdir/kexec_initrd_crypttab_path.txt"
+fi
+rm -rf /tmp/initrd_extract	
+
+if [ ! -z "$crypttab_entries" ];then
+	#insert current default's boot's initrd crypttab override, only modifying secret location
+	echo "The original crypttab options were overwritten to include secret.key: $crypttab_entries"	
+	echo "Adding modified entries under $bootdir/kexec_initrd_crypttab_entries.txt..."	
+	echo "$crypttab_entries" > "$bootdir/kexec_initrd_crypttab_entries.txt"
+else
+	echo "No crypttab entries found in extracted initrd. Removing $bootdir/kexec_initrd_crypttab_entries.txt"
```

Comment:
Same verbage suggestions as above, and here it might be more important to `rm -f "$bootdir/kexec_initrd_crypttab_*"` since we are leaving around a `..._path.txt` currently.  Maybe we could refactor this to consolidate the two copies of the failure handling?